Src/AutoFakeItEasy/AutoFakeItEasyCustomization.cs

```diff
@@ -41,11 +63,40 @@ public void Customize(IFixture fixture)
         {
             if (fixture == null) throw new ArgumentNullException(nameof(fixture));
 
+            if (this.GenerateDelegates)
+            {
+                fixture.Customizations.Add(new FakeItEasyRelay(new IsDelegateSpecification()));
+            }
+
             fixture.Customizations.Add(
                 new FakeItEasyBuilder(
                     new MethodInvoker(
                         new FakeItEasyMethodQuery())));
             fixture.ResidueCollectors.Add(this.Relay);
         }
+
+        private static void AssertFakeItEasyCanFakeDelegates()
+        {
+            var minimumFakeItEasyAssemblyVersion = new Version(1, 7, 4257, 42);
+            var actualFakeItEasyAssemblyVersion = typeof(A).GetTypeInfo().Assembly.GetName().Version;
+            if (actualFakeItEasyAssemblyVersion < minimumFakeItEasyAssemblyVersion)
+            {
+                throw new ArgumentException(String.Format(
+                    CultureInfo.CurrentCulture,
+                    "Option {0} was specified, but this requires FakeItEasy version {1} or higher, and {2} was detected. Either remove the option or upgrade FakeItEasy.",
+                    nameof(GenerateDelegates),
+                    minimumFakeItEasyAssemblyVersion,
+                    actualFakeItEasyAssemblyVersion));
+            }
+        }
+
+        private class IsDelegateSpecification : IRequestSpecification
```

Comment:
Please move this type to the `AutoFixture` assembly, `AutoFixture/Kernel` namespace. It's quite generic specification and it might sense to have it in the common library.

Also please add a few tests for it, if possible.

Reply:
Will do.